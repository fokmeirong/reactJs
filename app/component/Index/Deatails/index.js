import React from 'react';
import { connect } from "react-redux";
import { fetchMoviesDetailsById } from "../../../action/movieDetailsAction"

class Details extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        this.props.dispatch(fetchMoviesDetailsById(this.props.match.params.id));
    }

    render() {
        const { data } = this.props;
        return (
            <div className="film-details">
                <div className="detail-left">
                    <img src={ data.poster_path }/>
                </div>
                <div className="detail-right">
                    <p>
                        <span className="movie-name" >{ data.title }</span>
                        <span className="movie-rate" >{ data.vote_average }</span>
                    </p>
                    <p>
                        <span className="movie-overview" >{data.genres.join(" & ")}</span>
                    </p>
                    <p>
                        <span className="movie-release-date" >{ data.release_date }</span>
                    </p>
                    <p>
                        <span className="movie-description" >{ data.overview }</span>
                    </p>
                </div>
            </div>
        )
    }
}


const mapStateToProps = state => {
    return {
        data: state.detailsReducers.details
    };
};

export default connect(mapStateToProps)(Details);