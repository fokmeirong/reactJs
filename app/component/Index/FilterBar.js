import React from 'react';
import QueryStringManager from "../../../client/js/QueryStringManager"

class FilterBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            total: 0,
            sortOrder: "desc",
            sortBy:  QueryStringManager.get("sortBy") || "release_date",
        }
        this.handleSortbyClick = this.handleSortbyClick.bind(this);
    }

    handleSortbyClick(value) {
        this.setState({
            sortBy: value
        });
        QueryStringManager.set("sortBy", value);
        QueryStringManager.set("sortOrder", this.state.sortOrder);
        QueryStringManager.save();
    }

    render() {
        this.state.total = this.props.total;
        return (
            <div className="filter-bar">
                <div className="lfloat">{this.state.total} movies found</div>
                <div className="rfloat">
                    <span>Sort by</span>
                    <span className={"condition " + (this.state.sortBy === "release_date" ? "selected" : "") }>
                        <a onClick={() => this.handleSortbyClick("release_date")} >release date</a>
                    </span>
                    <span className={"condition " + (this.state.sortBy === "vote_count" ? "selected" : "") }>
                        <a onClick={() => this.handleSortbyClick("vote_count")}>rating</a>
                    </span>
                </div>
            </div>
        );
    }
}

export default FilterBar;