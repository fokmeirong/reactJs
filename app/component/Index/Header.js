import React from 'react';
import { connect } from "react-redux";
import QueryStringManager from "../../../client/js/QueryStringManager";
import { fetchMovies } from "../../action/moviesActions";
import { history } from '../../storeConfig';


class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            search: QueryStringManager.get("search"),
            searchBy: QueryStringManager.get("searchBy") || "title"
        }

        this.handleChange = this.handleChange.bind(this);
        this.handleSearchByClick = this.handleSearchByClick.bind(this);
        this.handleSearchButtonClick = this.handleSearchButtonClick.bind(this);
        this.handleKeyPress = this.handleKeyPress.bind(this);
    }

    handleChange(e) {
        this.setState({
            search: e.target.value
        });
    }

    handleSearchByClick(value) {
        this.setState({
            searchBy: value
        });
        QueryStringManager.set("searchBy", value);
    }

    handleSearchButtonClick() {
        QueryStringManager.set("search", this.state.search);
        QueryStringManager.set("searchBy", this.state.searchBy);
        // history.push("/Search/" + QueryStringManager.stringify());
        history.push("/Search/123");
        this.props.dispatchMovies(this.props.match.params.criteria);
    }

    handleKeyPress(target) {
        if (target.charCode == 13) {
            this.handleSearchButtonClick();
        }
    }

    render() {
        return (
            <header>
                <div className="logo">netflixroulette</div>
                <div>FIND YOUR MOVIE </div>

                <div className="search-bar">
                    <div className="movie-name">
                        <input className="movie-title" value={this.state.search} onChange={this.handleChange} onKeyPress={this.handleKeyPress} />
                    </div>
                    <div className="lfloat search-filter" >
                        <span>SEARCH BY</span>
                        <span className="search-by">
                            <a onClick={() => this.handleSearchByClick("title")} className={this.state.searchBy === "title" ? "selected" : ""} >TITLE</a>
                            <a onClick={() => this.handleSearchByClick("genres")} className={this.state.searchBy === "genres" ? "selected" : ""} >GENDER</a>
                        </span>
                    </div>
                    <div className="rfloat">
                        <button onClick={() => this.handleSearchButtonClick()}  >SEARCH</button>
                    </div>
                </div>
            </header>
        );
    }
}


const mapDispatchToProps = dispatch => ({
    dispatchMovies: () => dispatch(fetchMovies())
});

export default connect(null, mapDispatchToProps)(Header);