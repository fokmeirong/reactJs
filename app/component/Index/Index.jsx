
import React from 'react';
import createClass from 'create-react-class';

class Index extends React.Component {
    render() {
        return (
            <div>
                <Component1 name="1" />
                <Component2 name="2" />
                <Component3 name='3' />
                <Component4 name="4" />
            </div>
        );
    }
}

class Component1 extends React.Component {
    render() {
        return (
            <div>Hello World {this.props.name}</div>
        );
    }
}

function Component2(props) {
    return (
        <div>Hello World {props.name}</div>
    );
}

var Component3 = createClass({
    render() {
        return (
            <div>Hello World {this.props.name}</div>
        );
    }
});


class Component4 extends React.PureComponent {
    render() {
        return (
            <div>Hello World {this.props.name}</div>
        );
    }
};


export default Index;