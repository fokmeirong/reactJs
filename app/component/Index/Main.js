import React from 'react'
import Routers from './routers'

const Main = () => (
  <main>
    <Routers />
  </main>
)

export default Main