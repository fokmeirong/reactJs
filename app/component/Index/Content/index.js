import React from 'react';
import { connect } from "react-redux";
import Movies from './Movies/Movies';
import FilterBar from '../FilterBar'
import { fetchMovies } from "../../../action/moviesActions"

class Content extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        if (this.props.movies.length === 0) {
            return (
                <div className="no-films">No films found</div>
            )
        } else {
            return (
                <div>
                    <FilterBar total={this.props.total} />
                    <Movies movies={this.props.movies} />
                </div>
            )
        }
    }
}

const mapStateToProps = state => {
    return {
        movies: state.moivesReducers.movies,
        total: state.moivesReducers.total,
        loading: state.moivesReducers.loading,
        error: state.moivesReducers.error
    };
};


export default connect(mapStateToProps)(Content);