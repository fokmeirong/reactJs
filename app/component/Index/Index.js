
import React from 'react';
import Header from './Header';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary.js';
import Main from './Main';

export class Index extends React.Component {
  render() {
    return (
      <ErrorBoundary>
        <div>
          <Header />
          <Main />
        </div>
      </ErrorBoundary>
    );
  }
}

export default Index;