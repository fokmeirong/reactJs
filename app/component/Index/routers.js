import React from 'react'
import { Switch, Route } from 'react-router-dom'
import Content from './Content'
import Details from './Deatails'
import Blank from './Content/Blank'

export default () => {
    return <Switch>
        <Route exact path='/' component={Blank} />
        <Route path='/Search/:criteria' component={Content} />
        <Route path='/film/:id' component={Details} />
    </Switch>
}