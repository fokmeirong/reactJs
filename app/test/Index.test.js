import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Index from '../component/Index/Index.js';


describe('Index', () => {
    it('render Index page', () => {
      const wrapper = render(
        <Index/>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    })
});