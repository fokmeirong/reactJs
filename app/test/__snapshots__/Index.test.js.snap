// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index render Index page 1`] = `
<div>
  <header>
    <div
      class="logo"
    >
      netflixroulette
    </div>
    <div>
      FIND YOUR MOVIE
    </div>
    <div
      class="search-bar"
    >
      <div
        class="movie-name"
      >
        <input
          name="movie-title"
          placeholder="duck duck go"
        />
      </div>
      <div
        class="lfloat search-filter"
      >
        <span>
          SEARCH BY
        </span>
        <span
          class="search-by"
        >
          <a>
            TITLE
          </a>
          <a>
            GENDER
          </a>
        </span>
      </div>
      <div
        class="rfloat"
      >
        <button>
          SEARCH
        </button>
      </div>
    </div>
  </header>
  <div
    class="filter-bar"
  >
    <div
      class="lfloat"
    >
      3000 movies found
    </div>
    <div
      class="rfloat"
    >
      <span>
        Sort by
      </span>
      <span>
        <a>
          release date
        </a>
      </span>
      <span>
        <a>
          rating
        </a>
      </span>
    </div>
  </div>
  <div>
    <ul
      class="movies"
    >
      <li
        class="movie"
      >
        <div>
          <div
            class="movie-poster"
          >
            <img
              src="https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg"
            />
          </div>
          <div
            class="movie-descript"
          >
            <span>
              Gemini
            </span>
            <span>
              2018
            </span>
          </div>
          <div
            class="movie-genres"
          >
            Mystery & Thriller
          </div>
        </div>
      </li>
      <li
        class="movie"
      >
        <div>
          <div
            class="movie-poster"
          >
            <img
              src="https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg"
            />
          </div>
          <div
            class="movie-descript"
          >
            <span>
              Dilwale Dulhania Le Jayenge
            </span>
            <span>
              1995
            </span>
          </div>
          <div
            class="movie-genres"
          >
            Comedy & Drama & Romance
          </div>
        </div>
      </li>
      <li
        class="movie"
      >
        <div>
          <div
            class="movie-poster"
          >
            <img
              src="https://image.tmdb.org/t/p/w500/lt03HzS8tfDg5ipUQcLqIRNs0sC.jpg"
            />
          </div>
          <div
            class="movie-descript"
          >
            <span>
              Tisay
            </span>
            <span>
              2016
            </span>
          </div>
          <div
            class="movie-genres"
          >
            Drama & Romance
          </div>
        </div>
      </li>
      <li
        class="movie"
      >
        <div>
          <div
            class="movie-poster"
          >
            <img
              src="https://image.tmdb.org/t/p/w500/jwJoURyfm4XxtnYRtIOS2pYR9Np.jpg"
            />
          </div>
          <div
            class="movie-descript"
          >
            <span>
              Héctor El Father: Conocerás la verdad
            </span>
            <span>
              2018
            </span>
          </div>
          <div
            class="movie-genres"
          >
            Drama
          </div>
        </div>
      </li>
      <li
        class="movie"
      >
        <div>
          <div
            class="movie-poster"
          >
            <img
              src="https://image.tmdb.org/t/p/w500/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg"
            />
          </div>
          <div
            class="movie-descript"
          >
            <span>
              The Shawshank Redemption
            </span>
            <span>
              1994
            </span>
          </div>
          <div
            class="movie-genres"
          >
            Drama & Crime
          </div>
        </div>
      </li>
      <li
        class="movie"
      >
        <div>
          <div
            class="movie-poster"
          >
            <img
              src="https://image.tmdb.org/t/p/w500/rPdtLWNsZmAtoZl9PK7S2wE3qiS.jpg"
            />
          </div>
          <div
            class="movie-descript"
          >
            <span>
              The Godfather
            </span>
            <span>
              1972
            </span>
          </div>
          <div
            class="movie-genres"
          >
            Drama & Crime
          </div>
        </div>
      </li>
      <li
        class="movie"
      >
        <div>
          <div
            class="movie-poster"
          >
            <img
              src="https://image.tmdb.org/t/p/w500/xq1Ugd62d23K2knRUx6xxuALTZB.jpg"
            />
          </div>
          <div
            class="movie-descript"
          >
            <span>
              Your Name.
            </span>
            <span>
              2016
            </span>
          </div>
          <div
            class="movie-genres"
          >
            Romance & Animation & Drama
          </div>
        </div>
      </li>
      <li
        class="movie"
      >
        <div>
          <div
            class="movie-poster"
          >
            <img
              src="https://image.tmdb.org/t/p/w500/wKCkyb7dyDTbZn7qy6zh8oqtFrT.jpg"
            />
          </div>
          <div
            class="movie-descript"
          >
            <span>
              Sister's Younger Husband
            </span>
            <span>
              2016
            </span>
          </div>
          <div
            class="movie-genres"
          >
            Comedy & Drama & Romance
          </div>
        </div>
      </li>
      <li
        class="movie"
      >
        <div>
          <div
            class="movie-poster"
          >
            <img
              src="https://image.tmdb.org/t/p/w500/bVq65huQ8vHDd1a4Z37QtuyEvpA.jpg"
            />
          </div>
          <div
            class="movie-descript"
          >
            <span>
              The Godfather: Part II
            </span>
            <span>
              1974
            </span>
          </div>
          <div
            class="movie-genres"
          >
            Drama & Crime
          </div>
        </div>
      </li>
      <li
        class="movie"
      >
        <div>
          <div
            class="movie-poster"
          >
            <img
              src="https://image.tmdb.org/t/p/w500/5YUYg5q7QfC4IoNwNUtiwdiYKPr.jpg"
            />
          </div>
          <div
            class="movie-descript"
          >
            <span>
              Love, Simon
            </span>
            <span>
              2018
            </span>
          </div>
          <div
            class="movie-genres"
          >
            Comedy & Drama & Romance
          </div>
        </div>
      </li>
    </ul>
  </div>
  <footer>
    netflixroulette
  </footer>
</div>
`;
